components/server/src/theia-plugin/theia-plugin-service.ts

```diff
@@ -35,6 +36,11 @@ export class TheiaPluginService {
     @inject(StorageClient) protected readonly storageClient: StorageClient;
     @inject(TheiaPluginDB) protected readonly pluginDB: TheiaPluginDB;
     @inject(UserStorageResourcesDB) protected readonly userStorageResourcesDB: UserStorageResourcesDB;
+    protected readonly openVSXCachingProxyURL: string;
+
+    public constructor() {
+        this.openVSXCachingProxyURL = new GitpodHostUrl(process.env.HOST_URL).withDomainPrefix(localOpenVSXDomainPrefix).toString();
+    }
```

Comment:
This bit I find most interesting. OpenVSX blocking workspace starts is a no-go.
@akosyakov with VSCode resolving its plugins itself, other than for Theia, do we even still need this?